### PR TITLE
gh-109017: Use non alternate name for Kyiv

### PR DESCRIPTION
**Pull request by jspricke**
tzdata provides Kiev as an alternative to Kyiv:

https://sources.debian.org/src/tzdata/2023c-10/backward/?hl=314#L314

But Debian moved it to the tzdata-legacy package breaking the test:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1050530

This patch switches to the name provided by tzdata.

Also check that the new name is actually available.

* Issue: gh-109017

### CODE REVIEW

File: Lib/test/test_email/test_utils.py [hunk 1]
```diff
@@ -145,10 +146,9 @@ def test_localtime_epoch_notz_daylight_false(self):
     # XXX: Need a more robust test for Olson's tzdata
```
Comment (serhiy-storchaka) [inline Lib/test/test_email/test_utils.py]: And perhaps these three lines could be removed?

Comment (jspricke) [reply Lib/test/test_email/test_utils.py]: done, thx.

File: Lib/test/test_email/test_utils.py [hunk 2]
```diff
@@ -142,13 +143,9 @@ def test_localtime_epoch_notz_daylight_false(self):
...
         self.assertEqual(t1, t2)
 
-    # XXX: Need a more robust test for Olson's tzdata
-    @unittest.skipIf(sys.platform.startswith('win'),
-                     "Windows does not use Olson's TZ database")
-    @unittest.skipUnless(os.path.exists('/usr/share/zoneinfo') or
-                         os.path.exists('/usr/lib/zoneinfo'),
-                         "Can't find the Olson's TZ database")
-    @test.support.run_with_tz('Europe/Kiev')
+    @unittest.skipUnless("Europe/Kyiv" in zoneinfo.available_timezones()
```
Comment (serhiy-storchaka) [inline Lib/test/test_email/test_utils.py]:
```suggestion
    @unittest.skipUnless("Europe/Kyiv" in zoneinfo.available_timezones(),
```

**Comment by serhiy-storchaka** [comment]
Very well. Tested on Windows, so the check works.

Now I'm waiting for our buildbots to be fixed (they're not very stable at the moment), then I'll run tests with this PR on the buildbots, and if they pass, I'll merge the PR.

**Pull request merged**